gdal/frmts/netcdf/netcdfsg.cpp

```diff
@@ -0,0 +1,1047 @@
+					}	
+
+					// Figure out each Polygon's space requirements
+					if(noInteriors)
+					{
+						for(int ss = 0; ss < rc; ss++)
+						{
+						 	wkbSize += 16 * pnc[ss] + 1 + 4 + 4 + 4;
+						}
+					}
+
+					else
+					{
+						// Total space requirements for Polygons:
+						// (1 + 4 + 4) * number of Polygons
+						// 4 * number of Rings Total
+						// 16 * number of Points
+		
+
+						// Each ring collection corresponds to a polygon
+						wkbSize += (1 + 4 + 4) * poly_count[featureInd]; // (headers)
+
+						// Add header requirements for rings
+						wkbSize += 4 * parts_count[featureInd];
+
+						// Add requirements for number of points
+						wkbSize += 16 * nc;
+					}
+
+					// Now allocate and serialize
+					ret = new uint8_t[wkbSize];
+				
+					// Create Multipolygon headers
+					void * worker = (void*)ret;
+					worker = memcpy_jump(worker, &header, 1);
+					worker = memcpy_jump(worker, &t, 4);
+
+					if(noInteriors)
+					{
+						int cur_point = seek_begin;
+						size_t pcount = pnc.size();
+						worker = memcpy_jump(worker, &pcount, 4);
+
+						for(size_t itr = 0; itr < pcount; itr++)
+						{
+							worker = inPlaceSerialize_PolygonExtOnly(this, pnc[itr], cur_point, worker);
+							cur_point = pnc[itr] + cur_point;
+						}
+					}
+
+					else
+					{
+						int32_t polys = poly_count[featureInd];
+						worker = memcpy_jump(worker, &polys, 4);
+	
+						int base = pnc_bl[featureInd]; // beginning of parts_count for this multigeometry
+						int seek = seek_begin; // beginning of node range for this multigeometry
+						size_t ir_base = base + 1;
+						int rc_m = 1; 
+
+						// has interior rings,
+						for(int32_t itr = 0; itr < polys; itr++)
+						{	
+							rc_m = 1;
+
+							// count how many parts belong to each Polygon		
+							while(ir_base < int_rings.size() && int_rings[ir_base])
+							{
+								rc_m++;
+								ir_base++;	
+							}
+
+							if(rc_m == 1) ir_base++;	// single polygon case
+
+							std::vector<int> poly_parts;
+
+							// Make part node count sub vector
+							for(int itr_2 = 0; itr_2 < rc_m; itr_2++)
+							{
+								poly_parts.push_back(pnode_counts[base + itr_2]);
+							}
+
+							worker = inPlaceSerialize_Polygon(this, poly_parts, rc_m, seek, worker);
+
+							// Update seek position
+							for(size_t itr_3 = 0; itr_3 < poly_parts.size(); itr_3++)
+							{
+								seek += poly_parts[itr_3];
+							}
+						}
+					}
+				}	
+				break;
+
+				default:
+
+					throw new SG_Exception_BadFeature();	
+					break;
+		}
+
+		return ret;
+	}
+
+	void SGeometry_PropertyScanner::open(int container_id)
+	{
+		// First check for container_id, if variable doesn't exist error out
+		if(nc_inq_var(this->nc, container_id, NULL, NULL, NULL, NULL, NULL) != NC_NOERR)
+		{
+			return;	// change to exception
+		}
+
+		// Now exists, see what variables refer to this one
+		// First get name of this container
+		char contname[NC_MAX_NAME];
```

Comment:
+1 and 0 initialize ?

Reply:
ooh thanks I didn't see that :+1: will fix